packages/core/src/__tests__/acceptance/extension-point.acceptance.ts

```diff
@@ -0,0 +1,131 @@
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  BINDING_METADATA_KEY,
+  Context,
+  Getter,
+  MetadataInspector,
+} from '@loopback/context';
+import {expect} from '@loopback/testlab';
+import {addExtension, extensionPoint, extensions} from '../..';
+
+describe('extension point', () => {
+  describe('@extensionPoint', () => {
+    it('specifies name of the extension point', () => {
+      @extensionPoint('greeters')
+      class GreetingService {
+        @extensions()
+        public greeters: Getter<Greeter[]>;
+      }
+
+      expect(
+        MetadataInspector.getClassMetadata(
+          BINDING_METADATA_KEY,
+          GreetingService,
+        ),
+      ).to.eql({tags: {name: 'greeters'}});
+    });
+  });
+
+  describe('@extensions', () => {
+    let ctx: Context;
+
+    beforeEach(givenContext);
+
+    it('injects a getter function of extensions', async () => {
+      @extensionPoint('greeters')
+      class GreetingService {
+        @extensions()
+        public greeters: Getter<Greeter[]>;
+      }
+
+      ctx.bind('greeter-service').toClass(GreetingService);
+      registerGreeters('greeters');
+      const greeterService = await ctx.get<GreetingService>('greeter-service');
+      const greeters = await greeterService.greeters();
+      const languages = greeters.map(greeter => greeter.language);
+      expect(languages).to.containEql('en');
+      expect(languages).to.containEql('zh');
+    });
+
+    it('injects extensions based on the name tag of the extension point binding', async () => {
+      class GreetingService {
+        @extensions()
+        public greeters: Getter<Greeter[]>;
+      }
+      ctx
+        .bind('greeter-service')
+        .toClass(GreetingService)
+        .tag({name: 'greeters'}); // Tag the extension point with a name
+      registerGreeters('greeters');
+      const greeterService = await ctx.get<GreetingService>('greeter-service');
+      const greeters = await greeterService.greeters();
+      const languages = greeters.map(greeter => greeter.language);
+      expect(languages).to.containEql('en');
+      expect(languages).to.containEql('zh');
+    });
+
+    it('injects extensions based on the class name of the extension point', async () => {
+      class GreetingService {
+        @extensions()
+        public greeters: Getter<Greeter[]>;
+      }
+      ctx.bind('greeter-service').toClass(GreetingService);
+      registerGreeters(GreetingService.name);
+      const greeterService = await ctx.get<GreetingService>('greeter-service');
+      const greeters = await greeterService.greeters();
+      const languages = greeters.map(greeter => greeter.language);
+      expect(languages).to.containEql('en');
+      expect(languages).to.containEql('zh');
+    });
+
+    it('injects extensions based on extension point name from @extensions', async () => {
+      class GreetingService {
+        @extensions('greeters')
```

Comment:
Please add a test to demonstrate & verify what happens when there are two `@extensions()` properties on the same class.

```ts
class GreetingService {
  @extensions('greeters')
  public greeters(): Getter<Greeter[]>;

  @extensions('loggers')
  public loggers(): Getter<Logger[]>;
}
```